extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/MSQWorkerTaskLauncher.java

```diff
@@ -221,6 +225,17 @@ public List<String> getActiveTasks()
    */
   public void launchTasksIfNeeded(final int taskCount) throws InterruptedException
   {
+    // Fetch the list of workers which failed without work orders
+    Iterator<Integer> iterator = failedInactiveWorkers.iterator();
```

Comment:
This has to be in the synchronized (taskIds) block since multiple threads can call  `launchTasksIfNeeded`

Reply:
Moved